lib/layout/data_grid.dart

```diff
@@ -159,7 +163,19 @@ class EnsembleDataRow extends StatefulWidget
   @override
   Map<String, Function> setters() {
     return {
-      'visible': (dynamic v) => visible = Utils.getBool(v, fallback: visible)
+      'visible': (dynamic v) {
+        final bool newValue = Utils.getBool(v, fallback: visible);
+
+        // checking if old value and new value are different
+        if (visible != newValue) {
+          visible = newValue;
+
+          if (onVisibleChanged != null) {
+            // Firing the callback to update the state in EnsembleDataGrid
+            onVisibleChanged!(visible);
```

Comment:
this event is not needed. Can you just set the visibility on the widget and simply call setState? Wouldn't that be enough?